server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -49,21 +50,56 @@ protected boolean searchWithCollector(
         boolean hasFilterCollector,
         boolean hasTimeout
     ) throws IOException {
-        return searchWithCollectorManager(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
+        // create the top docs collector
+        final TopDocsCollectorContext topDocsFactory = createTopDocsCollectorContext(searchContext, hasFilterCollector);
+        return searchWithCollector(
+            searchContext,
+            searcher,
+            query,
+            collectors,
+            topDocsFactory,
+            hasFilterCollector,
+            hasTimeout,
+            topDocsFactory.shouldRescore()
+        );
+    }
+
+    protected boolean searchWithCollector(
+        SearchContext searchContext,
+        ContextIndexSearcher searcher,
+        Query query,
+        LinkedList<QueryCollectorContext> collectors,
+        QueryCollectorContext queryCollectorContext,
+        boolean hasFilterCollector,
+        boolean hasTimeout,
+        boolean shouldRescore
+    ) throws IOException {
+        return searchWithCollectorManager(
+            searchContext,
+            searcher,
+            query,
+            collectors,
+            queryCollectorContext,
+            hasFilterCollector,
+            hasTimeout,
+            shouldRescore
+        );
     }
 
     private static boolean searchWithCollectorManager(
         SearchContext searchContext,
         ContextIndexSearcher searcher,
         Query query,
         LinkedList<QueryCollectorContext> collectorContexts,
+        QueryCollectorContext queryCollectorContext,
         boolean hasFilterCollector,
-        boolean timeoutSet
+        boolean timeoutSet,
+        boolean shouldRescore
```

Comment:
I don't think you need `shouldRescore` anymore

Reply:
Yup, removed `shouldRescore` argument but forgot to add to commit. Now it should be fixed